lua/lualine/config.lua

```diff
@@ -6,6 +6,265 @@ local modules = require('lualine_require').lazy_require {
+---@field globalstatus boolean
+---@field refresh LualineRefresh
+
+---@alias LualineComponents
+---| "branch"
+---| "buffers"
+---| "diagnostics"
+---| "diff"
+---| "encoding"
+---| "fileformat"
+---| "filename"
+---| "filesize"
+---| "filetype"
+---| "hostname"
+---| "location"
+---| "mode"
+---| "progress`"
+---| "tabs"
+
+---@class LualineGeneralComponent
+-- Must be written as array value
+---@field [1] LualineComponents
+-- Enables the display of icons alongside the component.
+---@field icons_enabled boolean
+-- Defines the icon to be displayed in front of the component.
+---@field icon string
+-- When a string is provided it's treated as component_separator.
+-- When a table is provided it's treated as section_separator.
+-- Passing an empty string disables the separator.
+---@field separator string|table
+-- Condition function, the component is loaded when the function returns `true`.
+---@field cond fun():boolean
+-- Defines a custom color for the component.
+-- example: { fg = '#ffaa88', bg = 'grey', gui='italic,bold' } | { fg = 204 } | "WarningMsg"
+---@field color string|table
+-- Specify what type a component is.
+---@field type string|table
+-- Adds padding to the left and right of components.
+---@field padding number
+-- Format function, formats the component's output.
+---@field fmt fun():string
+
+---@class LualineColors
+-- Color for active buffer.
+--@field active string|table
+-- Color for inactive buffer.
+--@field inactive string|table
+
+---@class LualineBuffer: LualineGeneralComponent
+-- Shows shortened relative path when set to false.
+---@field show_filename_only boolean
+-- Shows indicator when the buffer is modified.
+---@field show_modified_status boolean
+-- 0: Shows buffer name
+-- 1: Shows buffer index (bufnr)
+-- 2: Shows buffer name + buffer index (bufnr)
+---@field mode number
+-- Maximum width of buffers component.
+---@field max_length number|fun():number
+-- Shows specific buffer name for that filetype.
+---@field filetype_names table
+-- Buffers colors
+---@field buffers_color LualineColors
+
+---@class LualineDiagnosticsColors
+-- Diagnostics' error color.
+---@field error string|table
+-- Diagnostics' warn color.
+---@field warn string|table
+-- Diagnostics' info color.
+---@field info string|table
+-- Diagnostics' hint color.
+---@field hint string|table
+
+---@class LualineDiagnosticsSymbols
+-- Diagnostics' error symbol (icons, then no_icons).
+---@field error string
+-- Diagnostics' warn symbol (icons, then no_icons).
+---@field warn string
+-- Diagnostics' info symbol (icons, then no_icons).
+---@field info string
+-- Diagnostics' hint symbol (icons, then no_icons).
+---@field hint string
+
+---@class LualineDiagnostics: LualineGeneralComponent
+-- Table of diagnostic sources. Possible values:
+-- 'nvim_lsp', 'nvim_diagnostic', 'coc', 'ale', 'vim_lsp',
+-- fun():{error=error_cnt, warn=warn_cnt, info=info_cnt, hint=hint_cnt}
+---@field sources string[]|fun():table
```

Comment:
instead of string array make it an enum array. as only above stated string values are valid.